resource/modules/resource_match.cpp

```diff
@@ -2633,6 +2638,58 @@ static void params_request_cb (flux_t *h, flux_msg_handler_t *w,
         flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
 }
 
+/*
+ * Mark a vertex as up or down
+ */
+static void set_status_request_cb (flux_t *h, flux_msg_handler_t *w,
+                                   const flux_msg_t *msg, void *arg)
+{
+    const char *rp = NULL, *st = NULL;
+    std::string resource_path = "", status = "", errmsg = "";
+    std::shared_ptr<resource_ctx_t> ctx = getctx ((flux_t *)arg);
+    resource_pool_t::string_to_status sts = resource_pool_t::str_to_status;
+
+    if (flux_request_unpack (msg, NULL, "{s:s, s:s}",
+                                        "resource_path", &rp,
+                                        "status", &st) < 0){
+        if (flux_respond_error (h, msg, EINVAL, "malformed RPC") < 0)
+            flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
+        return;
```

Comment:
You could save the repetition of the `flux_respond_error` and `flux_log_error` by setting `errmsg` and using `goto error` like this block: https://github.com/flux-framework/flux-sched/blob/0b70113bdf7eef5d501815d50155bdf124b14bae/resource/modules/resource_match.cpp#L1863

```suggestion
        errmsg = "malformed RPC";
        goto error;
```
That would apply to the following blocks, too.

Reply:
I originally wrote this function that way, but the compiler complained that I was jumping over the initialization of some variables, and so it prevented me from using the `auto` type. Do you have any recommendations to avoid that?

Reply:
I've run into that a lot, too. You'll need to declare the variable before the `if` blocks. Sometimes that isn't feasible; if that's true in this case then we can merge this PR.

Reply:
Done! And your spacing comments addressed.